Add a new tutorial for the Statement editor (#584)
wso2/docs-choreo-dev: en/docs/tutorials/construct-statements.md

> # Construct Statements

Reviewer: Are we going to link to this tutorial via the statement editor documentation icon?
<img width="605" alt="statement editor guide" src="https://user-images.githubusercontent.com/9528121/187830711-d56bf319-f98e-4cca-a248-932bff13cbb8.png">
Was just wondering whether it makes sense to have a tutorial like this if the purpose is to provide reference info to guide users to build any kind of statement that they would want to construct.

Author: Added the initial steps for accessing Choreo for the version hosted from the Documentation site. Can have a separate version for the document linked from the UI without those steps.

Reviewer 2: We discuss a separate location @sachiniSam can we share the design.

Reviewer 3: Refer Frame 3 -> https://www.figma.com/file/Sj7NyIFdvvvVcEeGEeHTHt/Statement-Editor?node-id=7%3A2331

# Construct Statements

> The Statement Editor in the Choreo Web Editor allows you to construct Ballerina statements depending on the use case you want to build. It is an intuitive visual editor that is easy to use even without an in-depth knowledge of the [Ballerina Programming Language](https://ballerina.io/).

Reviewer 4: Do we know what Ballerina statements are?

Reviewer 4: Okay..I guess we are talking language syntax.

    Here, you are entering static text as a part of the variable value.

5. Click **Expressions**, and under the **Arithmetic** section, click the **Es + Ex** expression.

    !!! info
        The **Expressions** tab lists all the pre-defined templates that you can use to define complex expressions.

    ![Select expression template](../assets/img/statement-editor/select-expression-template.gif){.cInlineImage-half}

    In the `Es + Ex` expression template, `Es` refers to the existing expression you clicked and selected. `+ Ex` combines another expression with `Es`.

6. Click **Libraries** and search for `max`. In the filtered list, click  **lang.int:max**.

    !!! info
        The **Libraries** tab lists all the Ballerina standard and language libraries.

    ![Select library](../assets/img/statement-editor/select-library.gif){.cInlineImage-half}

    `lang.int:max` is a Ballerina standard library for deriving the maximum value.

     You will see some diagnostics generated for the library you selected. Move on to the next step to resolve them. 

7. Click the **int0:max(n,ns)** function. 

>      When you select a function or a method call, the **Parameters** tab opens by default. This tab displays information about the selected function or method call, including details about parameters, descriptions, and return type descriptions. On this tab, you can easily add parameters to the expression by selecting the check boxes for the relevant parameters.

Reviewer 4: ```suggestion
     When you select a function or a method call, the **Parameters** tab opens by default. This tab displays information about the selected function or method call, including details about parameters, descriptions, and return type descriptions. On this tab, you can easily add parameters to the expression by selecting the checkboxes for the relevant parameters.
```

# Construct Statements

> The Statement Editor in the Choreo Web Editor is a tool that allows you to construct Ballerina statements in your Choreo Editor even without an in-depth knowledge of the Ballerina Programming Language.

Reviewer: We seem to be using the terms **Choreo Web Editor** and **Choreo Editor** in the same sentence. Better to stick to Web Editor for consistency. 
Also, Should we consider rephrasing to avoid repeating Web Editor?
 Something similar to:
“The Statement Editor in the Choreo Web Editor allows you to construct Ballerina statements depending on the use case you want to build. It is an intuitive visual editor that is easy to use even without an in-depth knowledge of the Ballerina Programming Language.” 
We can perhaps link off to ballerina.io from Ballerina Programming Language.

Author: Changed as suggested.

It provides the following:

- Context-based suggestions

- Ballerina libraries (standard libraries and language libraries)

- Pre-defined expression templates

> - Documentation for functions and methods with the support to easily configure parameters

Reviewer: Should we just say:
Documentation for functions and methods to easily configure parameters

Author: Changed as suggested.

> This guide walks you through the steps to build the following statement using the Statement-Editor.

Reviewer: We can remove the hyphen in Statement-Editor.

Author: Done

```
string maxValue = "Max : " + int0:max(30, -89).toString();
```

The above is a variable statement named `maxValue` that derives the maximum value in a list of two integers and passes the value in string format (for example, as `"Max : 30"` ).

> To build the statement, follow these steps:

Reviewer: We should perhaps mention that the steps on starting with the Echo template and so on are just for the purpose of this tutorial to get to the statement Editor.

Author: Moved these steps to an info note.

1. Sign in to the Choreo Console at [https://console.choreo.dev](https://console.choreo.dev).

2. On the **Home** page, scroll to the **Get started with a template** section. On the **Echo Service** template, click **Get Started**.

    The REST API created via the template opens on a separate page.
   
3. Click **Edit Code**. The Web Editor opens.

4. On the low-code diagram, click **+**. Then click **Variable** in the **Add Constructs** pane that opens.

    ![Add variable statement](../assets/img/statement-editor/add-variable-statement.png){.cInlineImage-full}

    The **Variable** pane opens, displaying the syntax for a variable statement with **`<add-expression>`** selected by default.

> 5. Click the **var** type descriptor on the statement and click **string** in the **Suggestions** tab.

Reviewer: ```suggestion
5. Click the **var** type descriptor on the statement and click **string** on the **Suggestions** tab.
```

    !!! info
        The **Suggestions** tab provides contextual suggestions based on the selected item. The **Suggestions** tab currently lists variable type suggestions because you selected the type descriptor.    

    ![Change type descriptor](../assets/img/statement-editor/change-type-descriptor.gif){.cInlineImage-half}

    Here, you are specifying that the variable value is a string.

6. To edit the variable name, double-click **`variable`** and type `maxValue` in the text box.

    !!! info

>         Another way to make a part of a statement editable is by clicking on the required statement component and pressing  **Enter**/**Return**.

Reviewer: ```suggestion
        Another way to make a part of a statement editable is to click on the required statement component and press  **Enter**/**Return**.
```

    ![Edit variable name](../assets/img/statement-editor/edit-variable-name.gif){.cInlineImage-half}

7. Double click **`<add-expression>`** and type `"Max : "` in the text box.

    !!! info

>         To navigate from one item on the statement to another, you can also press **Tab** / **Shift** + **Tab**.

Reviewer: Should we use the term or instead of /

Author: Changed as suggested.

    ![Edit variable expression](../assets/img/statement-editor/edit-variable-expression.gif){.cInlineImage-half}

    Here, you are entering static text as a part of the variable value.

8. Click **Expressions**, and under the **Arithmetic** section, click the **Es + Ex** expression.

    !!! info
        The **Expressions** tab lists all the pre-defined templates that you can use to define complex expressions.

    ![Select expression template](../assets/img/statement-editor/select-expression-template.gif){.cInlineImage-half}

    In the `Es + Ex` expression template, `Es` refers to the existing expression you clicked and selected. `+ Ex` combines another expression with `Es`.

9. Click **Libraries** and search for `max`. In the filtered list, click  **lang.int:max**.

    !!! info

>         The **Libraries** tab lists all the Ballerina Standard and Language libraries.

Reviewer: ```suggestion
        The **Libraries** tab lists all the Ballerina standard and language libraries.
```

Reviewer: Should we link off to https://lib.ballerina.io/#stdlibs?

    ![Select library](../assets/img/statement-editor/select-library.gif){.cInlineImage-half}

    `lang.int:max` is a Ballerina standard library for deriving the maximum value.

     You will see some diagnostics generated for the library you selected. Move on to the next step to resolve them. 

10. Click the **int0:max(n,ns)** function. 

>      When you select a function or a method call, the **Parameters** tab opens by default. This tab displays information about the selected function or method call, including details about parameters, descriptions, and return type descriptions. In this tab, you can easily add parameters to the expression by selecting the check boxes for the relevant parameters.

Reviewer: ```suggestion
     When you select a function or a method call, the **Parameters** tab opens by default. This tab displays information about the selected function or method call, including details about parameters, descriptions, and return type descriptions. On this tab, you can easily add parameters to the expression by selecting the check boxes for the relevant parameters.
```

Reviewer: Should we rephrase the last sentence and perhaps simplify it?
Suggestion:
On this tab, you can easily select appropriate parameters to add to the expression.

     ![Select function](../assets/img/statement-editor/select-function.gif){.cInlineImage-half}

     !!! note

>          When a parameter requires a function or a method call to work, the Statement Editor selects its check box for that parameter by default and does not allow you to clear it.

Reviewer: is this sentence giving the right idea?
Shouldn't it be 
"When a function or a method call requires a particular parameter to work, the Statement Editor selects the appropriate parameter by default and does not allow you to clear it."

Author: Corrected as pointed out.

11. Change the **n** and **ns** parameter values to integers. To make each parameter value editable, you can double-click on it. Let's change them as follows:

     | **Parameter** | **Value** |
     |---------------|-----------|
     | **n**         | **`30`**  |
     | **ns**        | **`-89`** |

12. Select **int0:max(30,-89)** and click **Suggestions**. Click **toString()** in the suggestions list to convert the value that the expression derived to a string format.

     ![Convert to string](../assets/img/statement-editor/convert-to-string.gif){.cInlineImage-half}

     Note that the Statement Editor has cleared all the diagnostics messages, indicating that the statement is valid.

     You will see the variable statement to be as follows:

     ```
     string maxValue = "Max : " + int0:max(30, -89).toString();
     ```
    
Congratulations! You have constructed a valid statement via the Statement Editor!

> You can click **Save** and add the variable statement to the low-code diagram.

Reviewer: ```suggestion
You can click **Save** to add the variable statement to the low-code diagram.
```

> !!! tip

Reviewer: Should we move this tip to the top so that it will be useful to a user who would want to undo, redo, or delete a step while constructing the statement?

Author: Done
